domenetjenester/brevbestiller/src/main/java/no/nav/foreldrepenger/melding/brevmapper/brev/InnvilgelseEngangstønadDokumentdataMapper.java

```diff
@@ -42,35 +45,60 @@ public String getTemplateNavn() {
                 .build();
 
-        return new EngangsstønadInnvilgelseDokumentdata.Builder()
-                .felles(felles)
-                .revurdering(behandling.erRevurdering())
-                .førstegangsbehandling(behandling.erFørstegangssøknad())
-                .medhold(BehandlingMapper.erMedhold(behandling))
-                .innvilgetBeløp(beregningsresultat.getBeløp())
-                .klagefristUker(brevParametere.getKlagefristUker())
-                .død(erDød(dokumentFelles))
-                .fbEllerMedhold(erFBellerMedhold(behandling))
-                .kontaktTelefonnummer(dokumentFelles.getKontaktTlf())
-                .endretSats(0)
-                .build();
+        var innvilgelseDokumentDataBuilder = EngangsstønadInnvilgelseDokumentdata.ny()
+                .medFelles(felles)
+                .medRevurdering(behandling.erRevurdering())
+                .medFørstegangsbehandling(behandling.erFørstegangssøknad())
+                .medMedhold(BehandlingMapper.erMedhold(behandling))
+                .medInnvilgetBeløp(beregningsresultat.getBeløp())
+                .medKlagefristUker(brevParametere.getKlagefristUker())
+                .medDød(erDød(dokumentFelles))
+                .medFbEllerMedhold(erFBellerMedhold(behandling))
+                .medErEndretSats(false);
+
+        if (behandling.erRevurdering()) {
+            Long nyttInnvBeløp = finnEndretSats(behandling, beregningsresultat);
+
+            if (nyttInnvBeløp != 0) {
+                innvilgelseDokumentDataBuilder.medErEndretSats(true);
+                innvilgelseDokumentDataBuilder.medInnvilgetBeløp(nyttInnvBeløp);
+
+            }
+
+        }
+
+        return innvilgelseDokumentDataBuilder.build();
     }
 
     private boolean erDød(DokumentFelles dokumentFelles) {
-        return dokumentFelles.getSakspartPersonStatus().equalsIgnoreCase(PersonstatusKode.DOD.toString());
+        if (dokumentFelles.getSakspartPersonStatus() == null) {
+            return false;
+        } else {
+            return dokumentFelles.getSakspartPersonStatus().equalsIgnoreCase(PersonstatusKode.DOD.toString());
+        }
     }
 
     private boolean erFBellerMedhold(Behandling behandling) {
         return BehandlingMapper.utledBehandlingsTypeInnvilgetES(behandling).equals(BehandlingsTypeType.FOERSTEGANGSBEHANDLING)
                 || BehandlingMapper.utledBehandlingsTypeInnvilgetES(behandling).equals(BehandlingsTypeType.MEDHOLD);
     }
+
+    private Long finnEndretSats (Behandling behandling, BeregningsresultatES beregningsresultat) {
+        Optional<Behandling> originalBehandling = domeneobjektProvider.hentOriginalBehandlingHvisFinnes(behandling);
+        BeregningsresultatES originaltBeregningsresultat = originalBehandling.map(domeneobjektProvider::hentBeregningsresultatES).orElse(null);
+
+        return originaltBeregningsresultat !=null ? Math.abs( beregningsresultat.getBeløp() - originaltBeregningsresultat.getBeløp()) : 0L;
```

Comment:
Er originaltBeregningsresultat == null en gyldig tilstand, eller kan det tyde på at noe er feil, slik at vi bør kaste IllegalArgumentException / Utviklerfeil i stedet for å sette 0L?

Reply:
enig at det virker som en ugyldig tilstand, men det er jeg ikke sikker på.. ? får sjekke litt